contracts/SpokePool.sol

```diff
@@ -1621,6 +1667,118 @@ abstract contract SpokePool is
+        // way (no need to have funds on the destination).
+        // If this is a slow fill, we can't exit early since we still need to send funds out of this contract
+        // since there is no "relayer".
+        bool isSlowFill = relayExecution.slowFill;
+        address recipientToSend = relayExecution.updatedRecipient;
+        if (msg.sender == recipientToSend && !isSlowFill) return;
+
+        // If relay token is wrappedNativeToken then unwrap and send native token.
+        address outputToken = relayData.outputToken;
+        if (outputToken == address(wrappedNativeToken)) {
+            // Note: useContractFunds is True if we want to send funds to the recipient directly out of this contract,
+            // otherwise we expect the caller to send funds to the recipient. If useContractFunds is True and the
+            // recipient wants wrappedNativeToken, then we can assume that wrappedNativeToken is already in the
+            // contract, otherwise we'll need the user to send wrappedNativeToken to this contract. Regardless, we'll
+            // need to unwrap it to native token before sending to the user.
+            if (!isSlowFill) IERC20Upgradeable(outputToken).safeTransferFrom(msg.sender, address(this), amountToSend);
+            _unwrapwrappedNativeTokenTo(payable(recipientToSend), amountToSend);
+            // Else, this is a normal ERC20 token. Send to recipient.
+        } else {
+            // Note: Similar to note above, send token directly from the contract to the user in the slow relay case.
+            if (!isSlowFill) IERC20Upgradeable(outputToken).safeTransferFrom(msg.sender, recipientToSend, amountToSend);
+            else IERC20Upgradeable(outputToken).safeTransfer(recipientToSend, amountToSend);
+        }
+
+        // @TODO Can we remove the third param in handleAcrossMessage since we no longer support partial fills? Will
+        // require integrators sign-off.
+        bytes memory updatedMessage = relayExecution.updatedMessage;
+        if (recipientToSend.isContract() && updatedMessage.length > 0) {
+            AcrossMessageHandler(recipientToSend).handleAcrossMessage(
+                outputToken,
+                amountToSend,
+                true, // fillCompleted
+                msg.sender,
+                updatedMessage
+            );
+        }
+
+        emit FilledUSSRelay(
+            relayData.inputToken,
+            relayData.outputToken,
+            relayData.inputAmount,
+            relayData.outputAmount,
+            relayExecution.repaymentChainId,
+            relayData.originChainId,
+            relayData.depositId,
+            relayData.fillDeadline,
+            relayData.exclusivityDeadline,
+            relayData.exclusiveRelayer,
+            relayer,
+            relayData.depositor,
+            relayData.recipient,
+            relayData.message,
+            replacedSlowFillExecution
+        );
+
+        emit USSRelayExecution(
```

Comment:
Is this event new? Why two events? (Sorry if we've already discussed this and I'm forgetting)

Reply:
Because the normal FilledRelay event has too many params including these such that the params emitted would have to be packed into a struct which is not nice for integrators. Moreover these params emitted here are not as commonly used. In fact the relayer-v2 repo doesn't use them so I thought it'd be better to emit them separately. 

The trade off of course is gas cost

Reply:
I think another tradeoff is that it is pretty tricky to associate this second event with the info in the first and vice-versa, making the event much harder to use.

Reply:
Hmm so maybe I'll revisit packing these updated params into a structure that the relayer can otherwise ignore in normal behavior

Reply:
Consolidated the event a bit here d5b4023329fb05baa5d065332b6a124f09c7ecce